### PR TITLE
[3.6.0] add deprecated funtions

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

File: native/cocos/bindings/manual/jsb_conversions_spec.cpp [hunk 1]
```diff
@@ -42,6 +43,7 @@
 #include "scene/Fog.h"
 #include "scene/Shadow.h"
 #include "scene/Skybox.h"
+#include "v8/Object.h"
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_conversions_spec.cpp]: Don't invoke v8 specific header.
it's better to use `#include "bindings/jswrapper/Object.h"`

File: native/cocos/bindings/manual/jsb_conversions_spec.cpp [hunk 2]
```diff
@@ -144,63 +119,32 @@ enum class MathType {
...
-    obj->setProperty("x", se::Value(v.x));
-    obj->setProperty("y", se::Value(v.y));
-    obj->setProperty("type", se::Value(static_cast<uint32_t>(MathType::VEC2)));
-    ret->setObject(obj);
-
-    return true;
-}
-
-bool Vec3_to_seval(const cc::Vec3 &v, se::Value *ret) { // NOLINT(readability-identifier-naming)
+bool Quaternion_to_seval(const cc::Quaternion &v, se::Value *ret) { // NOLINT(readability-identifier-naming)
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_conversions_spec.cpp]: Could Quaternion_to_seval be deprecated?

File: native/cocos/bindings/manual/jsb_conversions_deprecated.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+}
+CC_DEPRECATED(3.6, "use native_to_se instead")
+inline bool long_to_seval(long v, se::Value *ret) { // NOLINT(readability-identifier-naming, google-runtime-int)
+    return nativevalue_to_se(v, *ret, nullptr);
+}
+CC_DEPRECATED(3.6, "use native_to_se instead")
+inline bool ulong_to_seval(unsigned long v, se::Value *ret) { // NOLINT(readability-identifier-naming, google-runtime-int)
+    return nativevalue_to_se(v, *ret, nullptr);
+}
+CC_DEPRECATED(3.6, "use native_to_se instead")
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_conversions_deprecated.h]: `use native_to_se instead` to `use nativevalue_to_se instead`

File: native/cocos/bindings/manual/jsb_conversions_spec.cpp (hunk 1 above)
Comment (PatriceJiang) [reply native/cocos/bindings/manual/jsb_conversions_spec.cpp]: fixed

File: native/cocos/bindings/manual/jsb_conversions_spec.cpp (hunk 2 above)
Comment (PatriceJiang) [reply native/cocos/bindings/manual/jsb_conversions_spec.cpp]: It's a new interface introduced in 3.6

**Review by dumganhar — APPROVED** [review]

File: native/cocos/bindings/manual/jsb_conversions_spec.cpp [hunk 4]
```diff
@@ -1521,32 +1310,113 @@ bool nativevalue_to_se(const cc::Data &from, se::Value &to, se::Object * /*unuse
 
 // NOLINTNEXTLINE(readability-identifier-naming)
```
Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_conversions_spec.cpp]:
```suggestion
// NOLINTNEXTLINE
```

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Pull request merged**